src/query.jl

```diff
@@ -366,6 +366,7 @@ format inferred from the file's extension and/or magic bytes.
 """
 function query(filename::AbstractString)
     _, ext = splitext(filename)
+    filename = abspath(filename)
```

Comment:
I haven't thought much about this admittedly, I just went with the version that results in the same behavior for all branches, which seemed like a good thing to me (and made the tests pass).
So I don't feel strongly about it and your argument makes sense. I guess you need this more, when working with Streams.